tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -720,6 +719,92 @@ def conditional_program(
     assert np.allclose(np.where(mask, a, b)[1:], out.array()[:-1])
 
 
+def test_conditional_tuple_1():
+    size = 10
+    mask = np_as_located_field(IDim)(np.zeros((size,), dtype=bool))
+    mask.array()[0 : (size // 2)] = True
+    a = np_as_located_field(IDim)(np.ones((size,)))
+    c = np_as_located_field(IDim)(np.zeros((size,)))
+    d = np_as_located_field(IDim)(np.zeros((size,)))
+
+    @field_operator
+    def conditional_tuple(
+        mask: Field[[IDim], bool], a: Field[[IDim], float64]
+    ) -> tuple[Field[[IDim], float64], Field[[IDim], float64]]:
+        return where(mask, (a, 3.0), (2.0, 7.0))
+
+    @program
+    def conditional_tuple_p(
+        mask: Field[[IDim], bool],
+        a: Field[[IDim], float64],
+        c: Field[[IDim], float64],
+        d: Field[[IDim], float64],
+    ):
+        conditional_tuple(mask, a, out=(c, d))
+
+    conditional_tuple_p(mask, a, c, d, offset_provider={})
+
+    assert np.allclose(
+        np.where(mask, (a, np.full(size, 3.0)), (np.full(size, 2.0), np.full(size, 7.0))), (c, d)
+    )
+
+
+def test_conditional_tuple_2():
+    size = 10
+    mask = np_as_located_field(IDim)(np.zeros((size,), dtype=bool))
+    mask.array()[0 : (size // 2)] = True
+    a = np_as_located_field(IDim)(np.ones((size,)))
+    b = np_as_located_field(IDim)(np.ones((size,)))
+    c = np_as_located_field(IDim)(np.zeros((size,)))
+    d = np_as_located_field(IDim)(np.zeros((size,)))
+
+    @field_operator
+    def conditional_tuple_2_field_op(
+        mask: Field[[IDim], bool], a: Field[[IDim], float64], b: Field[[IDim], float64]
+    ) -> tuple[tuple[Field[[IDim], float64], Field[[IDim], float64]], Field[[IDim], float64]]:
+        return where(mask, ((3.0, 6.0), a), ((2.0, 7.0), b))
```

Comment:
shouldn't the result be checked here?

Reply:
Just for clarity: if the test is useful, but the result of execution is not important, the test should be a parsing test or lowering test, but not in `test_execution`.